File: 206. Reverse Linked List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        cursor = head

        while cursor:
            temp = cursor.next
            cursor.next = prev
            prev = cursor
            cursor = temp
    
        return prev

File: 206. Reverse Linked List/solution.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = NULL;
        ListNode* cursor = head;
        ListNode* temp;
        while(cursor != NULL)
        {
            temp = cursor->next;
            cursor->next = prev;
            prev = cursor;
            cursor = temp;
        }
        return prev;
}
};
